tests/Dialect/Secret/Conversions/secret_to_bgv/type_error.mlir

```diff
@@ -0,0 +1,100 @@
+// RUN: heir-opt --secret-to-bgv="poly-mod-degree=8" --verify-diagnostics %s
+
+// This example was converted from the dot_product_8f.mlir example via
```

Comment:
nit: maybe simplify this test case so that it's a bit more readable and instead add the dot_product as a high level expected-error test with `// RUN: heir-opt --mlir-to-bgv`? After all, we want to ensure that this error (and no other "spam" ones) trigger when people run the real pipelines

Reply:
Done